File: api/src/main/java/net/onelitefeather/pandorascluster/api/service/LandPlayerService.java
package net.onelitefeather.pandorascluster.api.service;

import net.onelitefeather.pandorascluster.api.enums.LandRole;
import net.onelitefeather.pandorascluster.api.land.LandArea;
import net.onelitefeather.pandorascluster.api.player.LandMember;
import net.onelitefeather.pandorascluster.api.player.LandPlayer;
import net.onelitefeather.pandorascluster.api.service.result.AddLandMemberResult;
import net.onelitefeather.pandorascluster.api.service.result.CreateLandPlayerResult;
import net.onelitefeather.pandorascluster.api.service.result.DeleteLandPlayerResult;
import net.onelitefeather.pandorascluster.api.service.result.RemoveLandMemberResult;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;
import java.util.UUID;

public interface LandPlayerService {

    /**
     * @param landArea the landArea of the land
     * @param member the member to add
     * @param landRole the land role
     */
    AddLandMemberResult addLandMember(@NotNull LandArea landArea, @NotNull LandPlayer member, @Nullable LandRole landRole);

    void updateLandMember(@NotNull LandArea landArea, @NotNull LandMember member);

    /**
     * @param member the member to remove
     */
    RemoveLandMemberResult removeLandMember(@NotNull LandMember member);

    @NotNull
    List<LandPlayer> getLandPlayers();

    /**
     * @param uuid the uuid of the player.
     * @param name the name of the player.
     */
    CreateLandPlayerResult createPlayer(@NotNull UUID uuid, @NotNull String name);

    /**
     * @param uuid the uuid of the player.
     */
    DeleteLandPlayerResult deletePlayer(@NotNull UUID uuid);

    @Nullable
    LandPlayer getLandPlayer(@NotNull UUID uuid);

    /**
     * @param uuid the uuid of the player.
     * @return true if the player exists
     */
    boolean playerExists(@NotNull UUID uuid);

    /**
     * @param landPlayer the player to update
     */
    void updateLandPlayer(@NotNull LandPlayer landPlayer);

}

File: api/src/main/kotlin/net/onelitefeather/pandorascluster/api/service/LandPlayerService.kt
package net.onelitefeather.pandorascluster.api.service

import net.onelitefeather.pandorascluster.api.land.Land
import net.onelitefeather.pandorascluster.api.enums.LandRole
import net.onelitefeather.pandorascluster.api.player.LandMember
import net.onelitefeather.pandorascluster.api.player.LandPlayer
import java.util.*

interface LandPlayerService {

    /**
     * @param land the land
     * @param member the member to add
     * @param landRole the land role
     */
    fun addLandMember(land: Land, member: LandPlayer, landRole: LandRole?)

    fun updateLandMember(land: Land, member: LandMember)

    /**
     * @param member the member to remove
     */
    fun removeLandMember(member: LandMember)

    fun getLandMember(land: Land, landPlayer: LandPlayer): LandMember?

    fun getLandPlayers(): List<LandPlayer>

    /**
     * @param uuid the uuid of the player.
     * @param name the name of the player.
     */
    fun createPlayer(uuid: UUID, name: String): Boolean

    /**
     * @param uuid the uuid of the player.
     */
    fun deletePlayer(uuid: UUID)

    /**
     * @param uuid the uuid of the player.
     * @return the land player by the uuid.
     */
    fun getLandPlayer(uuid: UUID): LandPlayer?

    /**
     * @param uuid the uuid of the player.
     * @return true if the player exists
     */
    fun playerExists(uuid: UUID): Boolean

    /**
     * @param landPlayer the player to update
     */
    fun updateLandPlayer(landPlayer: LandPlayer)

    /**
     * @param name the name of the player.
     * @return the land player by the name.
     */
    fun getLandPlayer(name: String): LandPlayer?
}